test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -577,6 +578,16 @@ Describe "Extended FileSystem Path/Location Cmdlet Provider Tests" -Tags "Featur
             $result = Split-Path -Path $level1_0Full -Leaf
             $result | Should Be $level1_0
         }
+        
+        It 'Validate LeafBase' {
+            $result = Split-Path -Path "$level2_1Full$fileExt" -LeafBase
+            $result | Should Be $level2_1
+        }
+
+        It 'Validate LeafBase' {
+            $result = Split-Path -Path "$level2_1Full$fileExt" -Extension
+            $result | Should Be $fileExt
+        }
```

Comment:
should there be explicit tests for things like:
"$level2_1Full$fileExt$fileExt"
to ensure that the trimming is not over-zealous?